index.d.ts

```diff
@@ -0,0 +1,60 @@
+export interface Action {
+  type: string;
```

Comment:
The only case I see for this is your example with `enum`:
```ts
enum ActionType {
   A, B, C
}
```
But where would it be useful? We can use it in reducer:
```ts
function reducer(state, action: Action<ActionType>) {
  // ...
}
```
But that would be incorrect: `action` argument here is not only your user-defined actions, it can also be `{type: '@@redux/INIT'}` or any other action used by third-party redux library.